src/lib/ThumbnailsSidebar.js

```diff
@@ -31,11 +42,41 @@ class ThumbnailsSidebar {
         this.anchorEl = element;
         this.pdfViewer = pdfViewer;
         this.thumbnailImageCache = {};
+        this.currentThumbnails = [];
 
         this.createPlaceholderThumbnail = this.createPlaceholderThumbnail.bind(this);
         this.requestThumbnailImage = this.requestThumbnailImage.bind(this);
         this.createThumbnailImage = this.createThumbnailImage.bind(this);
         this.generateThumbnailImages = this.generateThumbnailImages.bind(this);
+        this.thumbnailClickHandler = this.thumbnailClickHandler.bind(this);
+
+        this.anchorEl.addEventListener('click', this.thumbnailClickHandler);
+    }
+
+    /**
+     * Method to handle the click events in the Thumbnails Sidebar
+     *
+     * @param {Event} evt - Mouse click event
+     * @return {void}
+     */
+    thumbnailClickHandler(evt) {
+        const { target } = evt;
+
+        // Only care about clicks on the thumbnail element itself.
+        // The image and page number have pointer-events: none so
+        // any click should be the thumbnail element itself.
+        if (target.classList.contains(CLASS_BOX_PREVIEW_THUMBNAIL)) {
+            // Get the page number
+            const { bpPageNum: pageNumStr } = target.dataset;
```

Comment:
By binding to the anchor el, is there a chance we click on something that's not the thumbnail container? If `dataset` is undefined then trying to grab the page number would throw an exception.

Reply:
shouldn't be, but that's the if statement above so that only the element with the `bp-thumbnail` class will be processed as a click which should have the data attribute set for page number. Any other element would just be a `noop`

Reply:
good point 👍